### PR TITLE
Make GPT2 Dropout aligned with huggingface

### PR DESCRIPTION
**Pull request by xihui-wu**
This PR resolves issue https://github.com/tensorflow/swift-models/issues/104.

hugging face model for reference: https://github.com/huggingface/transformers/blob/2c1ebb8b507c19c75af5084b6c73e0b003c9eda6/src/transformers/modeling_gpt2.py

and dropout possibility settings:
https://github.com/huggingface/transformers/blob/2c1ebb8b507c19c75af5084b6c73e0b003c9eda6/src/transformers/configuration_gpt2.py#L122

### CODE REVIEW

File: Models/Text/GPT2/PythonCheckpointReader.swift [hunk 1]
```diff
@@ -129,6 +128,7 @@ extension TransformerLM: InitializableFromPythonCheckpoint {
         positionalEmbeddings = reader.readTensor(
             name: scope + "/wpe",
             scalarType: Float.self)
+        embeddingDropout = Dropout<Float>(probability: 0.1)
```
Comment (texasmichelle) [inline Models/Text/GPT2/PythonCheckpointReader.swift:131]: The type can be inferred here, no need to include it.

File: Models/Text/GPT2/TransformerLM.swift [hunk 2]
```diff
@@ -319,6 +320,7 @@ public struct TransformerLM: Module {
         let positionsTensor = Tensor<Int32>(shape: [1, tokens.shape[1]], scalars: positions)
         var h = embedding(tokens)
         h = h + positionalEmbeddings.gathering(atIndices: positionsTensor)
+        h = embeddingDropout(h)
```
Comment (texasmichelle) [inline Models/Text/GPT2/TransformerLM.swift:323]: This also belongs in the differentiable `callAsFunction()` below.

File: Models/Text/GPT2/TransformerLM.swift [hunk 3]
```diff
@@ -298,15 +296,18 @@ public struct EncoderLayer: Layer {
 public struct TransformerLM: Module {
     var embedding: Embedding<Float>
     var positionalEmbeddings: Tensor<Float>
+    var embeddingDropout: Dropout<Float> 
     var layers: [EncoderLayer]
     var norm: LayerNorm<Float>
 
     public init(
         embedding: Embedding<Float>, positionalEmbeddings: Tensor<Float>,
+        dropProbability: Double,
```
Comment (texasmichelle) [inline Models/Text/GPT2/TransformerLM.swift:305]: Why `Double` instead of `Float`?

File: Models/Text/GPT2/TransformerLM.swift [hunk 4]
```diff
@@ -298,15 +296,18 @@ public struct EncoderLayer: Layer {
...
         layers: [EncoderLayer], norm: LayerNorm<Float>
     ) {
         self.embedding = embedding
         self.positionalEmbeddings = positionalEmbeddings
+        self.embeddingDropout = Dropout<Float>(probability: dropProbability)
```
Comment (texasmichelle) [inline Models/Text/GPT2/TransformerLM.swift:310]: The type can be inferred here, too.

File: Models/Text/GPT2/TransformerLM.swift (hunk 2 above)
Comment (xihui-wu) [reply Models/Text/GPT2/TransformerLM.swift:323]: Ah! Thanks for catching this.

File: Models/Text/GPT2/TransformerLM.swift (hunk 3 above)
Comment (xihui-wu) [reply Models/Text/GPT2/TransformerLM.swift:305]: Dropout layer requires so: https://github.com/tensorflow/swift-apis/blob/5bc6e872f340095139d8b451281ec6aba182dec4/Sources/TensorFlow/Layers/Dropout.swift#L59

**Pull request merged**